GuardedView.fs

```diff
@@ -327,18 +367,30 @@ let SVGViewVars : GView<Sym<Var>> -> Set<TypedVar> =
     fun v ->
         let l = Multiset.toSet v
 
-        let symVarExprs =
-            function
-            | Bool e -> mapOverSymVars Mapper.mapBoolCtx findSymVars e
-            | Int e -> mapOverSymVars Mapper.mapIntCtx findSymVars e
+        let vars {Cond = g; Item = gf} =
+            let condvars = mapOverSymVars Mapper.mapBoolCtx findSymVars g
+            let itemvars = gfuncVars gf
 
-        let gfuncVars gf = Set.fold (+) Set.empty (Set.ofList (List.map symVarExprs gf.Params))
+            condvars
+            + itemvars
 
-        let vars(g, gf) =
-            mapOverSymVars Mapper.mapBoolCtx findSymVars g
-            + gfuncVars gf
+        Set.fold (+) Set.empty (Set.map vars l)
+
+let IteratedSVGViewVars : IteratedGView<Sym<Var>> -> Set<TypedVar> =
```

Comment:
I've changed the level of abstraction here so the functions operate over funcs and the callers do the heavy lifting to lift them over views.